### PR TITLE
Regenerate python bindings

### PR DESCRIPTION
**Pull request by v0lat1le**
I seem to have encountered the issue described in https://github.com/swig/swig/issues/567.

in python 3.5
``` python
from osgeo import ogr

def make_geom():
    geom = ogr.Geometry(ogr.wkbPoint)
    geom.AddPoint_2D(0, 0)
    return geom

def gen_list(N):
    for i in range(N):
        geom = make_geom()
        yield i

N = 10
print('gen', list(gen_list(N)))
```
produces `SystemError: <built-in function delete_Geometry> returned a result with an error set`

The issue have been resolved in swig 3.0.8 but the bindings that come with GDAL sources are generated with SWIG 2.0.12.

Regenerating the bindings with the latest swig resolves the issue, but also adds libpcre as a run-time dependency.

### CODE REVIEW

File: recipe/build.sh [hunk 1]
```diff
@@ -78,6 +78,12 @@ export CPPFLAGS="$CPPFLAGS -I$PREFIX/include"
...
             $OPTS
 
+# Regenerate python bindings.
+cd swig/python
+make veryclean
+make generate
+cd ../..
+
 # CircleCI offers two cores.
 make -j 2 >> $BUILD_OUTPUT 2>&1
```
Comment (ocefpaf) [inline recipe/build.sh]: Can you help me with another issue and change the `2` to `$CPU_COUNT` here? (`gdal` takes so long to build that I do not want to send a separate PR to fix that.)

**Comment by ocefpaf** [comment]
You need to bump the build number to get a new build uploaded on the channel.

**Comment by ocefpaf** [comment]
OS X is failing with:

```shell
sed: illegal option -- -
usage: sed script [-Ealn] [-i extension] [file ...]
       sed [-Ealn] [-i extension] [-e script] ... [-f script_file] ... [file ...]
/bin/sh: /bin/true: No such file or directory
```

and we don't have the Linux test for now due to a failure in `CircleCI`. I am testing Linux locally with our docker image. Will post the results soon.

**Comment by gillins** [comment]
Is it possible to do the same for Windows?

**Comment by v0lat1le** [comment]
I'm not sure what's up with the `sed` failure... Is it something I broke?

Don't have a local windows setup just yet, so I might have to abuse the CI while I get that going

**Comment by ocefpaf** [comment]
@v0lat1le can you add a test to ensure future updates won't break this again? You example in the first message is OK BTW.

**Comment by gillins** [comment]
> I'm not sure what's up with the sed failure... Is it something I broke?

I think OSX uses BSD `sed` rather than GNU `sed` and they have different options... I'm guessing this part of building `gdal` hasn't been tested on OSX. Is there a way of installing GNU `sed` with Conda? Or patching the makefile to work with BSD `sed`?

**Comment by ocefpaf** [comment]
@v0lat1le AppVeyor is stuck and we are trying to figure out what happened. Meanwhile do you mind adding the test `run_test.py` file?

**Comment by v0lat1le** [comment]
[GDAL updated](https://trac.osgeo.org/gdal/ticket/6749) the pre-generated bindings, so the next version (2.2+) will not have this specific issue.

**Comment by ocefpaf** [comment]
> [GDAL updated](https://trac.osgeo.org/gdal/ticket/6749) the pre-generated bindings, so the next version (2.2+) will not have this specific issue.

Thanks @v0lat1le! Even though that will be in the next release I don't think it hurts to always re-generate them here and test for consistency.

PS: Consider adding yourself as a maintainer to this feedstock :wink:

**Comment by ocefpaf** [comment]
Thanks @v0lat1le!
